services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/DocumentTableSchema.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Override
+    public AttributeValue attributeValue(EnhancedDocument item, String attributeName) {
+        if (item instanceof DefaultEnhancedDocument) {
+            return ((DefaultEnhancedDocument) item).getAttributeValueMap().get(attributeName);
+        }
+        throw new IllegalArgumentException("EnhancedDocument item is not instance of DefaultEnhancedDocument");
+    }
+
+    @Override
+    public TableMetadata tableMetadata() {
+        return tableMetadata;
+    }
+
+    @Override
+    public EnhancedType<EnhancedDocument> itemType() {
+        return EnhancedType.of(EnhancedDocument.class);
+    }
+
+    @Override
+    public List<String> attributeNames() {
+        return tableMetadata.primaryKeys().stream().collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean isAbstract() {
+        return false;
+    }
+
+    @NotThreadSafe
+    public static final class Builder {
+
+        private final StaticTableMetadata.Builder staticTableMetaDataBuilder = StaticTableMetadata.builder();
+
+        /**
+         * By Default the defaultConverterProvider is used for converting AttributeValue to primitive types.
+         */
+        private List<AttributeConverterProvider> attributeConverterProviders =
+            Collections.singletonList(ConverterProviderResolver.defaultConverterProvider());
+
+        /**
+         * Adds information about a partition key associated with a specific index.
+         *
+         * @param attributeName      the name of the attribute that represents the partition key
+         * @param attributeValueType the {@link AttributeValueType} of the partition key
+         */
+        public Builder primaryKey(String attributeName, AttributeValueType attributeValueType) {
+            staticTableMetaDataBuilder.addIndexPartitionKey(primaryIndexName(), attributeName, attributeValueType);
+            return this;
+        }
+
+        /**
+         * Adds information about a sort key associated with a specific index.
+         *
+         * @param attributeName      the name of the attribute that represents the sort key
+         * @param attributeValueType the {@link AttributeValueType} of the sort key
+         */
+        public Builder sortKey(String attributeName, AttributeValueType attributeValueType) {
+            staticTableMetaDataBuilder.addIndexSortKey(primaryIndexName(), attributeName, attributeValueType);
+            return this;
+        }
+
+        /**
+         * Specifies the {@link AttributeConverterProvider}s to use with the table schema. The list of attribute converter
+         * providers must provide {@link AttributeConverter}s for Custom types. The attribute converter providers will be loaded
+         * in the strict order they are supplied here.
+         * <p>
+         * If no AttributeConverterProvider are provided then  {@link DefaultAttributeConverterProvider} is used, which provides
```

Comment:
Suggesting rephrasing the paragraph to the following:
```
By default, {@link DefaultAttributeConverterProvider} will be used, 
and it will provide standard converters for most primitive and common Java types. 
Configuring this will override the default behavior, so it is recommended to 
always append `DefaultAttributeConverterProvider` when you configure the 
custom attribute converter providers.

<p>
sample code...
```